lib/datasource/go/index.ts

```diff
@@ -2,18 +2,19 @@ import URL from 'url';
 import { PLATFORM_TYPE_GITLAB } from '../../constants/platforms';
 import { logger } from '../../logger';
 import * as hostRules from '../../util/host-rules';
-import { Http } from '../../util/http';
 import { regEx } from '../../util/regex';
 import { trimTrailingSlash } from '../../util/url';
 import * as bitbucket from '../bitbucket-tags';
 import * as github from '../github-tags';
 import * as gitlab from '../gitlab-tags';
 import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';
+import { http } from './common';
+import * as goproxy from './goproxy';
+
+export { id } from './common';
 
-export const id = 'go';
 export const customRegistrySupport = false;
 
-const http = new Http(id);
 const gitlabRegExp = /^(https:\/\/[^/]*gitlab.[^/]*)\/(.*)$/;
```

Comment:
```suggestion
const gitlabRegExp = /^(https:\/\/[^/]*gitlab\.[^/]*)\/(.*)$/;
```

Should it match dot or any char? If yes, we can fix in separate pr.